File: web/src/main/java/no/nav/modiapersonoversikt/service/unleash/strategier/ByEnhetStrategy.java
package no.nav.modiapersonoversikt.service.unleash.strategier;

import no.finn.unleash.UnleashContext;
import no.finn.unleash.strategy.Strategy;

import java.util.Arrays;
import java.util.Map;

import static java.util.Optional.ofNullable;

public class ByEnhetStrategy implements Strategy {

    public static final String ENHETER = "enheter";

    @Override
    public String getName() {
        return "byEnhet";
    }

    @Override
    public boolean isEnabled(Map<String, String> parameters) {
        return false;
    }

    @Override
    public boolean isEnabled(Map<String, String> parameters, UnleashContext unleashContext) {

        return ofNullable(parameters)
                .map(par -> par.get("valgtEnhet"))
                .filter(s -> !s.isEmpty())
                .map(enheter -> enheter.split(","))
                .map(Arrays::stream)
                .map(enheter -> enheter.anyMatch(enhet -> ansattHarEnhet(unleashContext, enhet)))
                .orElse(false);
    }

    private boolean ansattHarEnhet(UnleashContext unleashContext, String enhet) {
        String enheter = unleashContext.getProperties().get(ENHETER) != null ? unleashContext.getProperties().get(ENHETER) : "";
        return Arrays.asList(enheter.split(",")).contains(enhet);
    }
}

File: web/src/main/java/no/nav/modiapersonoversikt/service/unleash/strategier/ByEnhetStrategy.kt
package no.nav.modiapersonoversikt.service.unleash.strategier

import io.getunleash.UnleashContext
import io.getunleash.strategy.Strategy
import no.nav.modiapersonoversikt.service.unleash.strategier.StrategyUtils.ENHETER_PROPERTY
import no.nav.modiapersonoversikt.service.unleash.strategier.StrategyUtils.splitIntoSet

class ByEnhetStrategy : Strategy {
    override fun getName(): String = "byEnhet"

    override fun isEnabled(parameters: Map<String, String>): Boolean {
        // Missing context, cannot verify "enhet"
        return false
    }

    override fun isEnabled(
        parameters: Map<String, String?>?,
        unleashContext: UnleashContext,
    ): Boolean {
        val strategiEnheter = parameters?.get(ENABLED_ENHETER_PROPERTY).splitIntoSet()
        val ansattesEnheter = unleashContext.properties[ENHETER_PROPERTY].splitIntoSet()

        return strategiEnheter.intersect(ansattesEnheter).isNotEmpty()
    }

    companion object {
        const val ENABLED_ENHETER_PROPERTY = "valgtEnhet"
    }
}
